services/common/src/redux/slices/complianceReportsSlice.spec.ts

```diff
@@ -0,0 +1,160 @@
+import { configureStore } from "@reduxjs/toolkit";
+import { notification } from "antd";
+import CustomAxios from "@mds/common/redux/customAxios";
+import complianceReportReducer, {
+  createMineReportDefinition,
+  fetchComplianceReports,
+  fetchMineReportDueDateTypes,
+} from "./complianceReportsSlice";
+import {
+  hideLoadingMock,
+  showLoadingMock,
+} from "@mds/common/redux/slices/mineReportPermitRequirementSlice.spec";
+
+// Mocks
+jest.mock("@mds/common/redux/customAxios");
+jest.mock("react-redux-loading-bar", () => ({
+  showLoading: () => showLoadingMock,
+  hideLoading: () => hideLoadingMock,
+}));
+jest.mock("@mds/common/redux/customAxios");
+
+jest.mock("react-redux-loading-bar", () => ({
+  showLoading: () => showLoadingMock,
+  hideLoading: () => hideLoadingMock,
+}));
+jest.mock("antd", () => ({
+  notification: {
+    success: jest.fn(),
+  },
+}));
+
+describe("complianceReportsSlice", () => {
+  let store;
+
+  // Mock responses
+  const mockComplianceReports = {
+    data: {
+      records: [
```

Comment:
probably a type on here- just so it gets flagged as an error if some of these properties change in the future.